packages/astro/src/types/public/content.ts

```diff
@@ -33,6 +33,33 @@ export interface MDXInstance<T extends Record<string, any>>
 	components: Record<string, AstroComponentFactory> | undefined;
 }
 
+/**
+ * Props accepted by the MDX `Content` component when using `await render()`.
+ * Allows passing custom components to override default MDX element rendering.
+ *
+ * @example
+ * ```astro
+ * ---
+ * import { getEntry, render } from 'astro:content';
+ * const entry = await getEntry('blog', 'post');
+ * const { Content } = await render(entry);
+ * ---
+ * <Content components={{ h1: MyCustomH1, img: MyImage }} />
+ * ```
+ */
+export interface MDXContentProps {
```

Comment:
Calling @ArmandPhilippot -- I don't think we document Markdown or MDX types, really. So, nothing to update, but is this something we should have?